### PR TITLE
feat(queries): Update Zig highlights scm

### PR DESCRIPTION
**Pull request by sbancuz**
Got these from nvim-treesitter
 - https://github.com/nvim-treesitter/nvim-treesitter

### CODE REVIEW

**Comment by sbancuz** [comment]
@jcs090218 can you look at this pr? Zig current highlights don't work with the new versions of the language. Also this may also close #624

**Comment by jcs090218** [comment]
Can you try to see if it works on your side? I'm pretty sure I got it from the nvim repo a while ago. 🤔

**Comment by sbancuz** [comment]
I'm not too sure what do you mean, but I can confirm that the highlights in my commit do work. They are not the best though. The old ones don't. Some keywords are not being highlighted 

image

If you want I can retry with the new upstream repo so this pr kill two birds with one stone, but I'm not sure on how to resolve the issues I'm having

**Comment by jcs090218** [comment]
Sorry for the late reply! I have been really busy recently!

I got the queries from the nvim repo, but there were errors so we couldn't use it directly. 🤔

If the CI passed, then it should be good! 👍

**Comment by sbancuz** [comment]
@jcs090218 I had to fight a bit with git submodules but in theory the CI should pass, I enabled it in my fork and it ran without any problems. This has the new repo and new-er highlights than the one from before. Although I didn't manage to test it on my machine since I kept getting the same problem of ABI incompatibility.

File: .gitmodules [hunk 1]
```diff
@@ -612,9 +606,9 @@
...
-[submodule "repos/gpr"]
-	path = repos/gpr
-	url = https://github.com/brownts/tree-sitter-gpr
+[submodule "repos/zig"]
+	path = repos/zig
+	url = https://github.com/tree-sitter-grammars/tree-sitter-zig.git
 	update = none
 	ignore = dirty
-	branch = main
+	branch = master
```
Comment (jcs090218) [inline .gitmodules:614]: I think you have accidentally delete the `grp` submodule?

**Comment by sbancuz** [comment]
Don't run the CI again because I think I saw some of the tests failing and I don't know why.

**Comment by jcs090218** [comment]
Just let me know if you need a re-run! :)

**Comment by sbancuz** [comment]
Ok my CI worked, so it **should** be all good I hope.

**Review by jcs090218 — APPROVED** [review]

**Comment by jcs090218** [comment]
Thank you for your hard work! 🚀

**Pull request merged**